src/diff/diff_test.go

```diff
@@ -109,6 +109,14 @@ func TestDiffNotEqual(t *testing.T) {
 
 func TestDiffEqualConfig(t *testing.T) {
 	projectState := createProjectState(t)
+
+	component := &model.Component{
+		ComponentKey: model.ComponentKey{
+			Id: "foo-bar",
+		},
+	}
+	projectState.Components().Set(component)
```

Comment:
Do kodu som pridal `ComponentsCache`, ... tu to vracia metoda`Components().
Je to nejaka kolekcia nacitanych komponent, ... ak je komponenta uz nacitana v pamati, tak sa uz opat nenacitava.
Ak nie je, tak sa nacita cez API.

Pri nacitani/ulozeni transformacie, potrebujem vediet, ze dana komponenta je typu `transformation`.
Tuto informaciu som ale nemal dostupnu tam, kde potrebujem, ... kedze `manifest.json` je iba `componentId`, nie `componentType`, to sa musi najprv nacitat cez API -> z toho vyplyvaju nasledujuce zmeny v kode.

`ComponentyType` si nechceme ukladat lokalne, ... potom by sme museli riesit, co robit, ak uzivatel zmeni hodnotu + to validovat.